lib/build.ts

```diff
@@ -121,6 +121,8 @@ export async function runBuildTask(
 	// which is why we check for 1.38 here
 	const usePlatformOption: boolean =
 		!!task.dockerPlatform &&
+		// workaround for `balena build` with Docker v20.10
+		task.dockerPlatform !== 'none' &&
```

Comment:
@robertgzr @dfunckt @toochevere, you may remember or be aware that last year we added the ability of selecting multiarch base images to `resin-multibuild` and this got deployed to `balena-builder` and became usable through `balena push`. But just before we got the CLI / `balena build` to use it, we found that it would break `balena build` when used with Docker v20.10. That very long thread: https://www.flowdock.com/app/rulemotion/i-cli/threads/RuSu1KiWOn62xaGy7O2sn8m8BUc

This change (and similar in `resolve.ts`) allows the CLI to bypass the "multiarch feature" (and keep `balena build` working without support for multiarch base images) if the CLI explicitly sets `task.dockerPlatform` to the string `'none'`, which I am doing in this other CLI PR:
https://github.com/balena-io/balena-cli/pull/2259